source/src_pw/charge.cpp

```diff
@@ -1220,6 +1220,7 @@ void Charge::cal_nelec(void)
 	// calculate number of bands (setup.f90)
 	//=======================================
 	double occupied_bands = static_cast<double>(nelec/ModuleBase::DEGSPIN);	
+	if(GlobalV::LSPINORB==1) occupied_bands = static_cast<double>(nelec);
```

Comment:
Why do we need a `static_cast` here?
Plus, you can try using ternary operator for this:
`double occupied_bands = static_cast<double>(GlobalV::LSPINORB==1 ? nelec : nelec/ModuleBase::DEGSPIN);`

Reply:
okay, thanks for you advice.

Reply:
> Why do we need a `static_cast` here?

👀 @dyzheng

Reply:
Sometimes, user would set odd number of electrons rather than even number, that would cause occupied_bands not be integer, we use a double type for dealing this situation.
Maybe better solution existed, but the solution in code is working well either.
So I suggest to accept this PR.